internal/ssh/ssh.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical.
+
+package ssh
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// JUJU_SSH_DEFAULT_PORT is the default port we expect the juju controllers to respond on.
+const JUJU_SSH_DEFAULT_PORT = 2223
+
+// Resolver is the interface with the methods needed by the ssh jump server to route request.
+type Resolver interface {
+	// GetAddrFromModelUUID is the method to resolve the address of the controller to contact given the model UUID.
+	GetAddrFromModelUUID(ctx context.Context, user openfga.User, modelUUID string) (string, error)
```

Comment:
I'd drop the Get prefix
```suggestion
	// AddrFromModelUUID is the method to resolve the address of the controller to contact given the model UUID.
	AddrFromModelUUID(ctx context.Context, user openfga.User, modelUUID string) (string, error)
```